nodes/core/storage/50-file.js

```diff
@@ -130,21 +135,42 @@ module.exports = function(RED) {
                             var packet = node.data.shift()
                             node.wstream.write(packet.data, function() {
                                 node.send(packet.msg);
+                                done();
                             });
 
                         } else {
                             // Dynamic filename - write and close the stream
                             var packet = node.data.shift()
                             node.wstream.end(packet.data, function() {
                                 node.send(packet.msg);
+                                delete node.wstream;
+                                delete node.wstreamIno;
+                                done();
                             });
-                            delete node.wstream;
-                            delete node.wstreamIno;
                         }
                     }
                 }
             }
+        }
+
+        this.on("input", function(msg) {
+            var msgQueue = node.msgQueue;
+            function processQ() {
```

Comment:
Can this be implemented without redefining the `processQ` function every time a message arrives?

Reply:
Moved `processQ` function outside of input callback function.